### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by APJohns**
Implement client-side search. This includes adding bootstrap to make use of existing search component.

### CODE REVIEW

File: src/styles/buttons-links.scss [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  opacity: 1;
+  cursor: not-allowed;
+}
+
+.icon-left {
+  margin-right: 5px;
+}
+
+.icon-right {
+  margin-left: 8px;
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Can we change this to 5px. Do you remember where you pulled this from?

File: src/styles/buttons-links.scss [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+    & i:nth-of-type(2n)
+    {
+        // for input addons with one button and two icons
+        // e.g.: date + time pickers with calendar and clock icons
+        padding-left: 8px;
+    }
+}
+
+.icon-btn {
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Can we change this to btn-icon to match base+modifier naming conventions?
(Sorry, this one is my fault.)

File: src/styles/buttons-links.scss [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+  &:focus {
+    position: absolute;
+    top: 0;
+    left: 0;
+    transition: top 0.5s ease-in;
+  }
+}
+
+.text-btn {
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Can we change this to btn-text to match base+modifier naming conventions?
(Sorry, this one is my fault.)

File: src/styles/buttons-links.scss [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+.btn-wide {
+  min-width: 150px;
+}
+
+.btn-block {
+  width: 100%;
+}
+
+.btn-group-toggle {
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Thoughts on pulling this out into toggle.scss?

File: src/styles/buttons-links.scss [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    }
+}
+
+.icon-btn {
+  width: 30px;
+  height: 30px;
+  padding: 5px;
+}
+
+.icon-btn-inline {
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Same re: base+modifier

File: src/styles/buttons-links.scss [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+  padding: 0;
+  color: $accent-color-primary;
+
+  &:hover {
+    background: transparent;
+    color: $accent-color-primary-hover;
+  }
+}
+
+.icon-btn-inline-tooltip {
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Same re: base+modifier
Thoughts on pulling this out into tooltips.scss?

File: src/styles/buttons-links.scss [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+      display: none;
+    }
+
+    & .fa-question {
+      color: $coconut;
+    }
+  }
+}
+
+.icon-btn[disabled] {
```
Comment (kaseyllabelle) [inline src/styles/buttons-links.scss]: Same re: base+modifier

File: src/styles/forms.scss [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+.invalid-feedback {
+    display: inline;
+    margin-top: 2px;
+    margin-bottom: 0;
+    color: $feedback-color-warning;
+    font-size: $font-size-body;
+}
+
+.input-validation-error,
```
Comment (kaseyllabelle) [inline src/styles/forms.scss]: We can remove this. This is specific to loyalty/kendo.

File: src/styles/forms.scss [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+.dropdown-validation-error {
+    border: 3px solid $feedback-color-warning !important; //TODO: verify if the important can be removed and also, if the changes in the border size are needed.
+}
+
+@media (min-width: 768px) {
```
Comment (kaseyllabelle) [inline src/styles/forms.scss]: This is specific to loyalty/kendo too. I want to find a better way to handle this in general, so let's remove it for now.

File: src/app/search/search.component.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+export class SearchComponent implements OnInit {
+  // Placeholder Text
+  @Input()
+  placeholder = "Search";
+
+  // Is the search is in progress?
+  @Input()
+  isSearching = false;
+
+  // Label for search field
```
Comment (kaseyllabelle) [inline src/app/search/search.component.ts]: Can we move the label above the placeholder so the code follows the UI top to bottom?

File: src/app/search/search.component.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<label for="search">{{ label }}</label>
```
Comment (kaseyllabelle) [inline src/app/search/search.component.html]: Is this an old file? Looks similar to src/app/table/table-filters/search/search.component.html

Comment (APJohns) [reply src/app/search/search.component.html]: Yep, forgot to remove the search when I moved it over, removing now

File: src/index.html [hunk 12]
```diff
@@ -8,8 +9,21 @@
   <link rel="icon" type="image/x-icon" href="favicon.ico">
   <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,600,700">
   <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"
```
Comment (kaseyllabelle) [inline src/index.html]: Remove

**Pull request merged**